packages/adblocker-extended-selectors/src/eval.ts

```diff
@@ -71,104 +71,210 @@ export function matches(element: Element, selector: AST): boolean {
       }
 
       return text.length >= minLength;
+    } else if (selector.name === 'matches-path') {
+      const { argument } = selector;
+      if (argument === undefined) {
+        return false;
+      }
+
+      const path = globalThis.window.location.pathname;
+
+      const pattern = argument.replace(/^\/|\/$/g, '');
+      const regex = new RegExp(pattern);
+
+      return regex.test(path);
+    } else if (selector.name === 'matches-attr') {
+      const { argument } = selector;
+      if (argument === undefined) {
+        return false;
+      }
+
+      const match = argument.match(/^([^=]+)="([^"]+)"$/);
+      if (!match) {
+        return false;
+      }
+
+      const [, attrName, pattern] = match;
+      const value = element.getAttribute(attrName);
+      if (value === null) {
+        return false;
+      }
+
+      const escapedPattern = pattern.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
```

Comment:
I think we have a utility for this already in network filters parsing code, maybe it's worth sharing?

Reply:
by looking at existing filters:
```
href=/[a-zA-Z0-9]{100,}/
/^on/=/event/
/-h?ref/
href=/https:\/\/thehackernews\.uk\/[a-zA-Z0-9]{4,}/
href=/^\/[-a-z]+\?[a-z]{2,}=/
href=/(^|audiobookbay\.lu)\/[-a-z0-9]+$/
class=/^[a-zA-Z]{2}$/
href="/__cft__\[0\]=[-\w]{265,}/"
class=/^[a-zA-Z]{2}$/
alt="/.*\shttps:\/\/t\.co\/[\w]{10}$/"
id=/[a-zA-Z]{40,}/
href="/__cft__\[0\]=[-\w]{290,}/"
href=/[a-zA-Z0-9]{100,}/
```
all of them should be handled as regexp without a need for escaping. New implementation does not have the escaping.